lib/elixir/test/elixir/enum_test.exs

```diff
@@ -417,6 +417,59 @@ defmodule EnumTest do
     assert Enum.into(["H", "i"], "") == "Hi"
   end
 
+  test "into/2 exceptions" do
+    item = 1
+    date = ~D[2015-01-01]
+    map = %{a: 1}
+    tuple = {:a, 1}
+
+    map_msg = "collecting into a map requires {key, value} tuples, got: #{inspect(item)}"
+    map_date_msg = "collecting into a map requires {key, value} tuples, got: #{inspect(date)}"
+    assert_raise ArgumentError, map_msg, fn -> Enum.into(Range.new(item, 10, 1), %{}) end
```

Comment:
You can directly write the range I think, it makes the test more readable (same below)
```suggestion
    assert_raise ArgumentError, map_msg, fn -> Enum.into(1..10, %{}) end
```